packages/round-manager/src/features/common/__tests__/AddQuestionModal.test.tsx

```diff
@@ -0,0 +1,148 @@
+import { fireEvent, render, screen } from "@testing-library/react";
+import { MemoryRouter } from "react-router-dom";
+import { CreateRoundContext, CreateRoundState, initialCreateRoundState } from "../../../context/round/CreateRoundContext";
+import { EditQuestion } from "../../api/types";
+import AddQuestionModal from "../AddQuestionModal";
+
+jest.mock("../../api/round");
+
+// test data for building out the questions
+const editQuestion: EditQuestion = {
+  index: 0,
+  field: {
+    title: "test",
+    required: false,
+    encrypted: false,
+    hidden: false,
+    inputType: "short-answer",
+    options: [],
+  },
+};
+
+beforeEach(() => {
+  jest.clearAllMocks();
+});
+
+describe("AddQuestionModal", () => {
+  beforeEach(() => {
+    // todo:
+  });
+
+  it("does not show the modal when the modal is not open", async () => {
+    renderWithContext(
+      <AddQuestionModal
+        show={false}
+        onClose={() => {
+          /**/
+        }}
+        onSave={() => {
+          /**/
+        }}
+        question={editQuestion}
+      />
+    );
+    
+    // expect(await screen.findByTestId("add-question-modal")).not.toBeVisible();
+  });
+  it("shows the new option row when the add option button is clicked", async () => {
+    renderWithContext(
+      <AddQuestionModal
+        show={true}
+        onClose={() => {
+          /**/
+        }}
+        onSave={() => {
+          /**/
+        }}
+        question={editQuestion}
+      />
+    );
+    expect(await screen.queryByTestId("add-question-modal")).toBeInTheDocument();
+
+    // todo: click the add option button and test that it added row for new option
+    const selectListButton = await screen.findByRole("select-list");
+    // fireEvent.change(selectListButton, { target: { value: "multi-select" } });
+    // fixme: this wont work until we can get the select list to open and select an option
+    // const addOptionButton = await screen.findByRole("add-option");
+    // fireEvent.click(addOptionButton);
+  });
+  it("shows proper question type when selected from menu", async () => {
+    // todo:
+    renderWithContext(
+      <AddQuestionModal
+        show={true}
+        onClose={() => {
+          /**/
+        }}
+        onSave={() => {
+          /**/
+        }}
+        question={editQuestion}
+      />
+    );
+    expect(await screen.queryByTestId("add-question-modal")).toBeInTheDocument();
+  });
+  it("shows two default options when a multi-select type is selected??", async () => {
+    // todo:
```

Comment:
are these tests still to be done or?

Reply:
@boudra yes we still need to finish the tests.